medusa/providers/generic_provider.py

```diff
@@ -781,6 +781,18 @@ def _get_title_and_url(self, item):
 
         return title, url
 
+    def _get_identifier(self, item):
```

Comment:
@medariox if you want to try it out. You'll have to copy this method into the provider, where you want to set the identifier. I can try it for binsearch. But you know more about the jacket stuff.